src/main/java/ui/issuepanel/PanelControl.java

```diff
@@ -191,15 +192,20 @@ public void closeCurrentPanel() {
         if (currentlySelectedPanel.isPresent()) {
             int panelIndex = currentlySelectedPanel.get();
             closePanel(panelIndex);
-            if (getChildren().size() == 0) {
-                setCurrentlySelectedPanel(Optional.empty());
-            } else {
-                int newPanelIndex = (panelIndex > getChildren().size() - 1)
-                                     ? panelIndex - 1
-                                     : panelIndex;
-                setCurrentlySelectedPanel(Optional.of(newPanelIndex));
-                getPanel(currentlySelectedPanel.get()).requestFocus();
-            }
+        }
+    }
+    
+    public void setFocusToPreviousPanel(int closedPanelIndex) {
+        if (getChildren().size() == 0) {
+            setCurrentlySelectedPanel(Optional.empty());
+        } else if (closedPanelIndex != currentlySelectedPanel.get()) {
+            return;
+        } else {
+            int newPanelIndex = (closedPanelIndex > getChildren().size() - 1)
+                                 ? closedPanelIndex - 1
+                                 : closedPanelIndex;
+            setCurrentlySelectedPanel(Optional.of(newPanelIndex));
+            getPanel(currentlySelectedPanel.get()).requestFocus();
```

Comment:
This method should be named in a more general way, since it doesn't always shift focus to the previous panel. `updateFocus`?